src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -391,7 +391,15 @@ public JsonPropertyInfo GetProperty(ReadOnlySpan<byte> propertyName, ref ReadSta
             return new Dictionary<string, JsonPropertyInfo>(capacity, comparer);
         }
 
-        public JsonPropertyInfo? PolicyProperty { get; private set; }
+        /// <summary>
+        /// The "type property" for the current JsonClassInfo.
+        /// A "type property" is not a real property on a type; instead it represents a collection element type,
+        /// a generic type parameter, or the root type passed into the root serialization APIs.
+        /// The "type property" is used to obtain the converter for the given type.
+        /// For example, for a property returning <see cref="Collections.Generic.List{T}"/> where T is a string,
+        /// a JsonClassInfo will be created with .Type=typeof(string) and .TypeProperty=JsonPropertyInfo{string}.
+        /// </summary>
+        public JsonPropertyInfo TypeProperty { get; private set; }
```

Comment:
I don't know if using `Type` is any better since that is such an overload term already (especially in a class like `JsonClassInfo`), but I don't have great alternatives. For this, a longer/more explicit name might be better. Maybe `ElementTypeProperty` or `PropertyHolderForElementTypeInfo`?

Reply:
It is used in places other than elements so "Element" shouldn't be in the name. For example, when we create a new stack frame for a POCO (due to a property returning that POCO) we will get the converter for that from the "TypeProperty".

Perhaps "PropertyInfoForClassInfo"?

Reply:
I think `PropertyInfoForClassInfo` is a good name.